v21.2/performance-best-practices-overview.md

```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
 
 - Use the [`SELECT FOR UPDATE`](select-for-update.html) statement in scenarios where a transaction performs a read and then updates the row(s) it just read. It orders transactions by controlling concurrent access to one or more rows of a table. It works by locking the rows returned by a [selection query](selection-queries.html), such that other transactions trying to access those rows are forced to wait for the transaction that locked the rows to finish. These other transactions are effectively put into a queue that is ordered based on when they try to read the value of the locked row(s).
 
-- When replacing values in a row, use [`UPSERT`](upsert.html) and
-  specify values for all columns in the inserted rows. This will
-  usually have the best performance under contention, compared to
-  combinations of [`SELECT`](select-clause.html),
-  [`INSERT`](insert.html), and [`UPDATE`](update.html).
-
-- Increase
-  [normalization](https://en.wikipedia.org/wiki/Database_normalization)
-  of the data to place parts of the same records that are modified by
-  different transactions in different tables. Note however that this
-  is a double-edged sword, because denormalization can also increase
-  performance by creating multiple copies of often-referenced data in
-  separate ranges.
-
-- If the application strictly requires operating on very few different
-  index key values, consider using [`ALTER ... SPLIT
-  AT`](split-at.html) so that each index key value can be served by
-  a separate group of nodes in the cluster.
-
-It is always best to avoid contention as much as possible via the
-design of the schema and application. However, sometimes contention is
-unavoidable. To maximize performance in the presence of contention,
-you'll need to maximize the performance of a single range.
-To achieve this, multiple strategies can be applied:
-
-- Minimize the network distance between the replicas of a range,
-  possibly using zone configs and partitioning.
+- When replacing values in a row, use [`UPSERT`](upsert.html) and specify values for all columns in the inserted rows. This will usually have the best performance under contention, compared to combinations of [`SELECT`](select-clause.html), [`INSERT`](insert.html), and [`UPDATE`](update.html).
+
+- Increase [normalization](https://en.wikipedia.org/wiki/Database_normalization) of the data to place parts of the same records that are modified by different transactions in different tables. Increasing normalization has drawbacks as well as benefits, because denormalization can also increase performance by creating multiple copies of often-referenced data in separate ranges.
```

Comment:
The second sentence is very hard to parse because it refers to both increasing normalization and denormalization.  It's not clear what the drawbacks and benefits are. 

Could you please revise it?

Reply:
I'll expand this, though part of me wonders why we're even raising counter-arguments to data normalization in a section about avoiding transaction contention.